src/Action/Platform/Nrps/EditMembershipAction.php

```diff
@@ -54,28 +54,24 @@ class EditMembershipAction
     /** @var RouterInterface */
     private $router;
 
-    /** @var MemberFactoryInterface */
-    private $memberFactory;
-
     public function __construct(
         FlashBagInterface $flashBag,
-        MembershipRepository $repository,
+        MembershipService $service,
         Environment $twig,
         FormFactoryInterface $factory,
         RouterInterface $router,
         MemberFactoryInterface $memberFactory
     ) {
         $this->flashBag = $flashBag;
-        $this->repository = $repository;
+        $this->service = $service;
         $this->twig = $twig;
         $this->factory = $factory;
         $this->router = $router;
-        $this->memberFactory = $memberFactory;
     }
 
     public function __invoke(Request $request, string $membershipIdentifier): Response
     {
-        $membership = $this->repository->find($membershipIdentifier);
+        $membership = $this->service->findMembership($membershipIdentifier);
```

Comment:
```suggestion
        $membership = $this->service->findMembership($membershipIdentifier, [MemberInterface::STATUS_DELETED]);
```

Reply:
That would fetch only deleted ones. I don't think this would be intentional.